### PR TITLE
Evenly distribute syntax nodes to different files, improving compile time

### PR DESCRIPTION
**Pull request by ahoppen**
This improves the compile time of swift-syntax in debug mode by ~2s (from ~11s to ~9s) by distributing the syntax node definitions evenly over multiple files.

### CODE REVIEW

File: CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxNodesFile.swift [hunk 1]
```diff
@@ -38,213 +38,216 @@ fileprivate extension Node {
...
-        try EnumDeclSyntax(
-          """
-          public enum \(name): RawSyntaxNodeProtocol
-          """
-        ) {
-          for (caseName, kind) in choices {
-            DeclSyntax("case `\(caseName)`(\(kind.rawType))")
-          }
+func rawSyntaxNodesFile(nodesStartingWith: [Character]) -> SourceFileSyntax {
+  SourceFileSyntax(leadingTrivia: copyrightHeader) {
```
Comment (kimdv) [inline CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxNodesFile.swift:42]: Should we have a return here?

File: CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift [hunk 2]
```diff
@@ -129,8 +128,12 @@ struct GenerateSwiftSyntax: ParsableCommand {
...
         GeneratedFileSpec(swiftSyntaxBuilderGeneratedDir + ["RenamedChildrenBuilderCompatibility.swift"], renamedChildrenBuilderCompatibilityFile),
       ]
-      + BASE_KIND_FILES.map { baseKind in
-        GeneratedFileSpec(swiftSyntaxGeneratedDir + ["syntaxNodes", baseKind.value], syntaxNode(emitKind: baseKind.key))
+      // This split of letters produces files for the syntax nodes that have about equal size, which improves compile time
+      + ["AB", "C", "D", "EF", "GHI", "JKLMN", "OP", "QRS", "TUVWXYZ"].flatMap { letters in
+        [
+          GeneratedFileSpec(swiftSyntaxGeneratedDir + ["syntaxNodes", "SyntaxNodes\(letters).swift"], syntaxNode(nodesStartingWith: Array(letters))),
+          GeneratedFileSpec(swiftSyntaxGeneratedDir + ["raw", "RawSyntaxNodes\(letters).swift"], rawSyntaxNodesFile(nodesStartingWith: Array(letters))),
+        ]
```
Comment (kimdv) [inline CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift:136]: Nice idea! 
Firstly I was "Why not each type get their own file" and I came across the cmake file 😬

**Review by kimdv — APPROVED** [review]

File: CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift (hunk 2 above)
Comment (kimdv) [reply CodeGeneration/Sources/generate-swiftsyntax/GenerateSwiftSyntax.swift:136]: I'm not that much into how the invitation of the Swift frontend work. 
But thanks for the explanation! 🙌

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
I did a mistake while rebasing and missed a guard based on the first letter in raw syntax generation. Fixed now

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by bnbarham — APPROVED** [review]
I suppose this is about as arbitrary split as decl/expr/etc anyway. Bit weird, but if it helps with compile time 🤷

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**